src/Map.ts

```diff
@@ -0,0 +1,465 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { EqualityComparer, OrderingComparer, Predicate } from "./prelude"
+import { Option } from "./Option"
+import { pipe } from "./composition"
+
+const defaultEqualityComparer: EqualityComparer<never> = {
+    equals: (a, b) => a === b,
+}
+
+const defaultOrderingComparer: OrderingComparer<never> = {
+    compare: (a, b) => {
+        const stringA = String(a)
+        const stringB = String(b)
+        return stringA === stringB ? 0 : stringA.localeCompare(stringB) < 0 ? -1 : 1
+    },
+    equals: defaultEqualityComparer.equals,
+}
+
+/**
+ * Lookup a key/value pair (wrapped in a `Some`) from a `Map` using the given key.
+ * If the `Map` doesn't contain the key, returns `None`.
+ *
+ * Uses the given equality comparer if passed. Otherwise, defaults to reference
+ * equality (triple equals) for equality comparisons.
+ *
+ * @category Lookups
+ *
+ * @returns An `Option` containing a tuple of the key and value.
+ */
+const findWithKey =
```

Comment:
:question: Wondering if this should be something like `tryGet` to make it more clear its returning an `Option` (the `try` part) and that is roughly equivalent to the built in `.get()` on `Map`.